pandas/tseries/index.py

```diff
@@ -740,6 +740,18 @@ def _add_offset(self, offset):
                            PerformanceWarning)
             return self.astype('O') + offset
 
+    def putmask(self, mask, value):
+        """
```

Comment:
Ah my explanation is incorrect. This is remained because of my early attempt, it can be removed.